platform/core/lib/src/core.rs

```diff
@@ -18,6 +19,7 @@ pub struct Core<'a> {
     #[allow(dead_code)]
     pub preferences: Arc<Preferences>,
```

Comment:
@GhenadiePusca wouldn't it make sense for `Preferences` to also be held in a mutex? That way we could make the `set_` methods work with `&self mut`. Plus, it's probably not thread safe right now, is it?

Reply:
I expect an object to be mutable if it stores any state, currently preferences writes and reads it state from the DB, also the DB handles the thread safety. It would be better to not have two ways of storing the state.